src/hooks/useEntitlement.ts

```diff
@@ -55,6 +57,7 @@ const useEntitlement: UseEntitlement = (playlistItem) => {
       queryFn: () => checkoutService?.getEntitlements({ offerId }, sandbox),
       enabled: !!playlistItem && !!user && !!offerId && !isPreEntitled,
       refetchOnMount: 'always' as const,
+      notifyOnChangeProps,
```

Comment:
Is there a reason we can't just use string literals here?
```suggestion
      ['data', 'isLoading'],
```

Reply:
Without the `as const` modifiers, the type results in `string[]` and is not accepted by react-query.

If I use `['data', 'isLoading'] as const` the type results in `readonly ['data', 'isLoading']` and is also not accepted by react-query.

This does work but is not much cleaner IMHO

```ts
const notifyOnChangeProps: QueryObserverOptions['notifyOnChangeProps'] = ['data', 'isLoading'];
```